### DISCUSSION TITLE
GDPR compliance when querying data

### DISCUSSION THREAD

**Discussion by NordlingDev** [supabase/supabase#9308] [Questions] [answered]
I just noticed that a simple `supabaseClient.from('users').select('*').eq('name', 'real name').single()` query will build this URL:

```
GET /rest/v1/users?select=name&name=eq.real+name
```

If these URLs are being logged anywhere or even turn up in browser history, it could potentially lead to GDPR issues. What are your thoughts?

**Comment by b2m9** [comment]
Not a lawyer, but did some GDPR things in the past. Here are my thoughts:

In terms of logging, yes, all logs are a bit of a problem with regards to GDPR. That being said, most companies I've worked for purge logs after 30 days unless another system intervenes (e.g. logging for security reasons). You can actually see that many companies have some fine print for GDPR deletion requests that it can take up to 30 days for the logs to be deleted as they usually have no system in place to go through logs.

In terms of browser history, I fail to see how this is relevant for GDPR as a user's browser is not a system the company is managing thus it is not part of your data processing responsiblity. I might be wrong here, happy to be corrected.

**Comment by chasers** [comment, marked as answer]
I built the logging stuff for Supabase (Logflare).

Yeah, browser history isn't an issue.

Generally speaking, there are concessions for logging in GDPR. 

We have something on the roadmap to let you control the level of logs across all services. Although we use these for some important observability metrics. Error rate per project, for example. So we'll need to make sure we don't mess this up. And all this is very useful for debugging issues. 

Supabase stores your Postgres logs for up to 90 days and the edge logs for up to 12 months. We need to store some for 12 months for SOC 2 compliance. If you need to handle an information delete request we can actually delete logs so if you contact support we'll do that for you. All logs live in the EU too.

Protecting the PII your system sees is very important. But, the fact that you store PII (anywhere, your own db included) is not the issue.  It’s:

- What you tell people you collect
- What you tell them you collect it for
- Do they agree to that (is this in your terms of service)
- And do you actually use it for that

People get in trouble when they collect PII for one reason and use if for another (sell it to ad companies).
